src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,545 @@
+     * recommended, but for example could be useful for testing with a self-signed certificate.
+     * <P>
+     * The SSL authentication is enabled by default meaning that hostname verification
+     * and certificate chain validation is done using the JVM default settings.
+     * </P>
+     *
+     * @return this ClientBuilder object for setting additional options
+     * @throws IllegalStateException if {@link #customSSLSocketFactory(SSLSocketFactory)}
+     * has been called on this ClientBuilder
+     * @see #customSSLSocketFactory(SSLSocketFactory)
+     */
+    public ClientBuilder disableSSLAuthentication() {
+        if(authenticatedModeSSLSocketFactory == null) {
+            this.isSSLAuthenticationDisabled = true;
+        } else {
+            throw new IllegalStateException("Cannot disable SSL authentication when a " +
+                    "custom SSLSocketFactory has been set.");
+        }
+        return this;
+    }
+    
+
+    /**
+     * Specifies the custom SSLSocketFactory to use when connecting to Cloudant over a
+     * <code>https</code> URL, when SSL authentication is enabled.
+     *
+     * @param factory An SSLSocketFactory, or <code>null</code> for the
+     *                default SSLSocketFactory of the JRE.
+     * @return this ClientBuilder object for setting additional options
+     * @throws IllegalStateException if {@link #disableSSLAuthentication()}
+     * has been called on this ClientBuilder
+     * @see #disableSSLAuthentication()
+     */
+    public ClientBuilder customSSLSocketFactory(SSLSocketFactory factory) {
+        if(!isSSLAuthenticationDisabled) {
+            this.authenticatedModeSSLSocketFactory = factory;
+        } else {
+            throw new IllegalStateException("Cannot use a custom SSLSocketFactory when " +
+                    "SSL authentication is disabled.");
+        }
+        return this;
+    }
+
+    /**
+     * This method adds {@link HttpConnectionInterceptor}s to be used on the CloudantClient
+     * connection. Interceptors can be used to modify the HTTP requests and responses between the
+     * CloudantClient and the server.
+     * <P>
+     * An example interceptor use might be to apply a custom authorization mechanism. For
+     * instance to use BasicAuth instead of CookieAuth it is possible to add a
+     * {@link com.cloudant.http.interceptors.BasicAuthInterceptor}:
+     * </P>
+     * <pre>
+     * {@code
+     * CloudantClient client = ClientBuilder.account("yourCloudantAccount")
+     *      .interceptors(new BasicAuthInterceptor("yourUsername:yourPassword"))
+     *      .build()
+     * }
+     * </pre>
+     *
+     * @param interceptors one or more HttpConnectionInterceptor objects
+     * @return this ClientBuilder object for setting additional options
+     * @see HttpConnectionInterceptor
+     */
+    public ClientBuilder interceptors(HttpConnectionInterceptor... interceptors) {
```

Comment:
I don't think its necessary, just a nice to have, I'm fine with leaving it with just a var args method, we can always add it later if users ask for it.